### DISCUSSION TITLE
A way to ignore conflicts among extras?

### DISCUSSION THREAD

**Discussion by erezsh** [python-poetry/poetry#6411] [Q&A] [answered]
My project, data-diff, supports a lot of "extra" 3rd party dependencies, i.e. packages that are only necessary in specific situations.

We tell users to install them if they wish, for example with `pip install data-diff[mysql, snowflake, bigquery]`

Some of these packages have conflicting dependencies, let's say some of them want `numpy==1.3.0`, and some want `numpy>=1.5.0`. That is causing `poetry update` to fail.

However, the conflicting packages are probably not going to be used as the same time. So poetry fails entirely, for a situation that would only happen <1% of the time!

My questions are -

- Is there a way to get poetry to ignore conflicts between certain packages?
- If yes, is there also a way to tell pip that certain combinations of dependencies aren't supported? (like, if someone chooses both snowflake and databricks, it gives an error saying they are incompatible, but each one can still be installed individually)
- Is there a different way to go about this? I could drop the "extras" feature from my pyproject, and tell the users to just install the 3rd party libraries directly, with `pip install mysql-connector snowflake-connector-python etc etc` . But then I have even less control over version compatibility. And it makes the installation harder.

**Comment by neersighted** [comment, marked as answer]
There is not a way to avoid conflicts between extras -- Poetry's solver is universal and solves for all possible permutations across extras, groups, and platforms. I'm not sure there's a good way to do what you want with Poetry right now, as it simply has not been designed (in its present state) for this advanced library use case.

Poetry has come a long way, but remains at its core a tool that was designed for application and simple to moderately complex library scenarios. I would invite you to open an issue to discuss if what you want is in scope for Poetry, and well as potential design patterns.
